### ISSUE TITLE
Upgrade kotlin dependency to 1.2.50

### ISSUE THREAD

**Issue by rgoldberg** [gradle/kotlin-dsl#927]
Upgrade kotlin dependency to 1.2.50

**Comment by mkobit** [comment]
I don't know if it is planned, but there have been a bunch of scripting improvements/changes at https://github.com/Kotlin/KEEP/blob/master/proposals/scripting-support.md

**Comment by rgoldberg** [comment]
Can Gradle’s Kotlin dependency be upgraded to 1.2.50, and later, over time, be made to take advantage of the improved scripting support, without requiring all scripting support to be switched over before the updated Kotlin can be used?

**Comment by bamboo** [comment]
Blocked by https://youtrack.jetbrains.com/issue/KT-24946

**Comment by bamboo** [comment]
Superseded by #940.
